anvil/src/eth/backend/fork.rs

```diff
@@ -93,6 +87,39 @@ impl ClientFork {
         self.storage.write()
     }
 
+    /// Sends `eth_call`
+    pub async fn call(
+        &self,
+        request: &CallRequest,
+        block: Option<BlockNumber>,
+    ) -> Result<Bytes, ProviderError> {
+        let tx = ethers::utils::serialize(request);
+        let block = ethers::utils::serialize(&block.unwrap_or(BlockNumber::Latest));
+        self.provider().request("eth_call", [tx, block]).await
```

Comment:
should this instead be `self.provider().call(tx, block).await`? can we dedup the `CallRequest` type with the `TypedTransaction`?

Reply:
this is a delegate of the RPC call, so assuming deserializing is lossless here this should be enough, without converting first, consuming the type.

the main reason this wasn't `TypedTransaction` in the first place is because of some `NameOrAddress` types that are not easy to work with down the line, and this ensures ethers' features don't change the request type